src/odemis/driver/orsay.py

```diff
@@ -0,0 +1,880 @@
+        """
+        if parameter is None:
+            parameter = self._system.TurboPump1.IsOn
+        if parameter is not self._system.TurboPump1.IsOn:
+            raise ValueError("Incorrect parameter passed to _updateTurboPumpOn. Parameter should be "
+                             "datamodel.HybridPlatform.PumpingSystem.TurboPump1.IsOn. Parameter passed is %s"
+                             % parameter.Name)
+        if attributeName != "Actual":
+            return
+        state = str(parameter.Actual).lower() == "true"
+        logging.debug("Turbopump turned %s." % ("on" if state else "off"))
+        self.turboPumpOn._set_value(state, force_write=True)
+
+    def _updatePrimaryPumpOn(self, parameter=None, attributeName="Actual"):
+        """
+        parameter (Orsay Parameter): the parameter on the Orsay server to use to update the VA
+        attributeName (str): the name of the attribute of parameter which was changed
+
+        Reads if the primary pump is currently on from the Orsay server and saves it in the primaryPumpOn VA
+        """
+        if parameter is None:
+            parameter = self.parent.datamodel.HybridPlatform.PrimaryPumpState
+        if parameter is not self.parent.datamodel.HybridPlatform.PrimaryPumpState:
+            raise ValueError("Incorrect parameter passed to _updatePrimaryPumpOn. Parameter should be "
+                             "datamodel.HybridPlatform.PrimaryPumpState. Parameter passed is %s"
+                             % parameter.Name)
+        if attributeName != "Actual":
+            return
+        state = str(parameter.Actual).lower() == "true"
+        logging.debug("Primary pump turned %s." % ("on" if state else "off"))
+        self.primaryPumpOn._set_value(state, force_write=True)
+
+    def _updateNitrogenPressure(self, parameter=None, attributeName="Actual"):
+        """
+        parameter (Orsay Parameter): the parameter on the Orsay server to use to update the VA
+        attributeName (str): the name of the attribute of parameter which was changed
+
+        Reads pressure on nitrogen inlet to the turbopump from the Orsay server and saves it in the nitrogenPressure VA
+        """
+        if parameter is None:
+            parameter = self._system.Manometer1.Pressure
+        if parameter is not self._system.Manometer1.Pressure:
+            raise ValueError("Incorrect parameter passed to _updateNitrogenPressure. Parameter should be "
+                             "datamodel.HybridPlatform.PumpingSystem.Manometer1.Pressure. Parameter passed is %s"
+                             % parameter.Name)
+        if attributeName == "Actual":
+            return
+        self.nitrogenPressure._set_value(float(parameter.Actual), force_write=True)
+
+    def terminate(self):
+        """
+        Called when Odemis is closed
+        """
+        self._system.Manometer1.ErrorState.Unsubscribe(self._updateErrorState)
```

Comment:
We have the convention that it's always safe to call .terminate() multiple times in a row. So please, protect with a `if self._system`.